Angular 6, rxjs 6 support

Angular 6 is released, I think this will be a small fix about rxjs6. 

 > ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/fromEvent' in '/Users/ismet.ozozturk/Desktop/mesele/mesele-frontend/node_modules/ngx-infinite-scroll/modules'
ERROR in ./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/of' in '/Users/ismet.ozozturk/Desktop/mesele/mesele-frontend/node_modules/ngx-infinite-scroll/modules'

### Environment
- [ ] Version used: 0.8.4

same error here.

./node_modules/ngx-infinite-scroll/modules/ngx-infinite-scroll.es5.js
Module not found: Error: Can't resolve 'rxjs/observable/fromEvent' in 'C:\Users\scketcher\Desktop\NoQAdmin\node_modules\ngx-infinite-scroll\modules'

--- a/package.json
+++ b/package.json
@@ -1,8 +1,7 @@
 {
   "name": "ngx-infinite-scroll",
   "version": "0.8.4",
-  "description":
-    "An infinite scroll directive for Angular compatible with AoT compilation and Tree shaking",
+  "description": "An infinite scroll directive for Angular compatible with AoT compilation and Tree shaking",
   "main": "./bundles/ngx-infinite-scroll.umd.js",
   "module": "./modules/ngx-infinite-scroll.es5.js",
   "es2015": "./modules/ngx-infinite-scroll.js",
@@ -29,26 +28,30 @@
     "url": "https://github.com/orizens/ngx-infinite-scroll/issues"
   },
   "homepage": "https://github.com/orizens/ngx-infinite-scroll",
-  "keywords": ["angular", "javascript", "typescript"],
+  "keywords": [
+    "angular",
+    "javascript",
+    "typescript"
+  ],
   "license": "MIT",
   "peerDependencies": {
-    "@angular/common": ">= 4.0.0",
-    "@angular/core": ">= 4.0.0"
+    "@angular/common": ">= 6.0.0",
+    "@angular/core": ">= 6.0.0"
   },
   "devDependencies": {
-    "@angular/animations": "^4.0.0",
-    "@angular/common": "^4.0.0",
-    "@angular/compiler": "^4.0.0",
-    "@angular/compiler-cli": "^4.0.0",
-    "@angular/core": "^4.0.0",
-    "@angular/platform-browser": "^4.0.0",
-    "@angular/platform-browser-dynamic": "^4.0.0",
-    "@angular/platform-server": "^4.0.0",
+    "@angular/animations": "^6.0.0",
+    "@angular/common": "^6.0.0",
+    "@angular/compiler": "^6.0.0",
+    "@angular/compiler-cli": "^6.0.0",
+    "@angular/core": "^6.0.0",
+    "@angular/platform-browser": "^6.0.0",
+    "@angular/platform-browser-dynamic": "^6.0.0",
+    "@angular/platform-server": "^6.0.0",
+    "@compodoc/compodoc": "^1.1.2",
     "@types/jasmine": "2.5.46",
     "@types/node": "7.0.10",
     "chalk": "1.1.3",
-    "codelyzer": "3.0.0-beta.4",
-    "compodoc": "0.0.41",
+    "codelyzer": "^4.3.0",
     "core-js": "2.4.1",
     "jasmine-core": "2.5.2",
     "karma": "1.5.0",
@@ -59,17 +62,17 @@
     "karma-webpack": "2.0.3",
     "reflect-metadata": "0.1.10",
     "rollup": "0.41.6",
-    "rxjs": "^5.5.6",
+    "rxjs": "^6.1.0",
     "shelljs": "0.7.7",
     "sorcery": "0.10.0",
     "ts-helpers": "1.1.2",
     "ts-loader": "2.0.3",
     "tslint": "4.5.1",
-    "typescript": "^2.1.5",
-    "uglify-js": "2.8.15",
+    "typescript": "2.7.2",
+    "uglify-js": "^2.8.15",
     "webpack": "2.3.1",
     "yargs": "7.0.2",
-    "zone.js": "0.8.5"
+    "zone.js": "^0.8.26"
   },
   "dependencies": {
     "opencollective": "^1.0.3"

--- a/spec.bundle.js
+++ b/spec.bundle.js
@@ -9,7 +9,7 @@ require('zone.js/dist/sync-test');
 require('zone.js/dist/jasmine-patch');
 require('zone.js/dist/async-test');
 require('zone.js/dist/fake-async-test');
-require('rxjs/Rx');
+require('rxjs');
 
 var testing = require('@angular/core/testing');
 var browser = require('@angular/platform-browser-dynamic/testing');

--- a/src/modules/infinite-scroll.directive.ts
+++ b/src/modules/infinite-scroll.directive.ts
@@ -10,7 +10,7 @@ import {
   Output,
   SimpleChanges
 } from '@angular/core';
-import { Subscription } from 'rxjs/Subscription';
+import {Subscription} from 'rxjs';
 
 import { InfiniteScrollEvent, IInfiniteScrollAction } from '../models';
 import { hasWindowDefined, inputPropChanged } from '../services/ngx-ins-utils';

--- a/src/services/ngx-ins-utils.ts
+++ b/src/services/ngx-ins-utils.ts
@@ -1,4 +1,3 @@
-import * as console from 'console';
 import { ElementRef, SimpleChange } from '@angular/core';
 
 export function resolveContainerElement(

--- a/src/services/scroll-register.ts
+++ b/src/services/scroll-register.ts
@@ -1,7 +1,4 @@
-import { ElementRef } from "@angular/core";
-import { Observable } from "rxjs/Observable";
-import { of } from "rxjs/observable/of";
-import { fromEvent } from "rxjs/observable/fromEvent";
+import { Observable, of, fromEvent } from "rxjs";
 import { map, mergeMap, tap, sampleTime, filter } from "rxjs/operators";
 
 import * as Models from "../models";

--- a/tests/services/scroll-register.spec.ts
+++ b/tests/services/scroll-register.spec.ts
@@ -1,12 +1,7 @@
 import * as Models from '../../src/models';
-import { Observable } from 'rxjs/Observable';
-import {
-  async,
-  inject
-} from '@angular/core/testing';
+import { Observable } from 'rxjs';
 import * as ScrollRegister from '../../src/services/scroll-register';
 import * as ScrollResolver from '../../src/services/scroll-resolver';
-import * as EventTrigger from '../../src/services/event-trigger';
 import { ElementRef } from '@angular/core';
 
 describe('Scroll Regsiter', () => {

